packages/theme/src/cli/services/pull.ts

```diff
@@ -149,6 +150,21 @@ async function executePull(theme: Theme, session: AdminSession, options: PullOpt
     return
   }
 
+  /**
+   * If the current directory is a Git directory and it is not clean, we ask for
+   * confirmation before proceeding.
+   */
+  if (
+    (await insideGitDirectory(path)) &&
+    !(await isClean(path)) &&
+    !(await currentDirectoryConfirmed(
+      force,
+      'The current Git directory has uncommitted changes. Do you want to proceed?',
+    ))
+  ) {
+    return
+  }
```

Comment:
I think we could restructure this a tiny bit. There's no need to do any of the checks if `force` is `true`:

```suggestion
  if (!force) {
    const dirtyDirectory = (await insideGitDirectory(path)) && !(await isClean(path));
    if (
      dirtyDirectory && 
      !(await currentDirectoryConfirmed(
        force,
        'The current Git directory has uncommitted changes. Do you want to proceed?',
      ))
    ) {
      return
    }
```

Still a little awkward but maybe a touch clearer?